src/Tasks/Engines/DoormanRunner.php

```diff
@@ -78,6 +78,10 @@ public function runQueue($queue)
         $descriptor = $this->getNextJobDescriptorWithoutMutex($queue);
 
         while ($manager->tick() || $descriptor) {
+            if (QueuedJobService::isMaintenanceLockActive()) {
+                return;
```

Comment:
Do you want to add some logging here, if there's a logger available? Just something like `->info('Skipped queued job descriptor since maintenance log is active')` maybe?